### ISSUE TITLE
Feature Request: Support for Ansible-Vault

### ISSUE THREAD

**Issue by zkanda** [vermut/intellij-ansible#33]

**Comment by dblockow** [comment]
Agree, support for vault would be fantastic: http://docs.ansible.com/ansible/playbooks_vault.html

**Comment by vermut** [comment]
Well, how do you imagine typical workflow? Automatic decoding? Or option to preview encoded data?

**Comment by zkanda** [comment]
I can imagine two workflow

* Creating and editing encrypted data
* Editing existing encrypted data
* Maybe Optional Storing of password? Not sure if this is a good practice, but ansible-vault can accept parameterized *--vault-password-file=PASSWORD_FILE*. Which you need to store in a file anyway.

**Comment by lhoss** [comment]
+1 (with ideas from Zkanda)
And allow to define a (default) vault-password-file, which would be used to (on-the-fly) encrypt/decrypt all vault managed(/encrypted) files

**Comment by foozmeat** [comment]
+1

**Comment by thebenwaters** [comment]
+1

**Comment by voorth** [comment]
+1

**Comment by lzedl** [comment]
+1

**Comment by jeffgrant** [comment]
+1

**Comment by jamesmmchugh** [comment]
+1

**Comment by redj45** [comment]
+1

**Comment by dcfsc** [comment]
I have a global password file defined in /etc/ansible

I would love to right-click a vault.yml file, Choose "Ansible Vault > Encrypt" and have it encrypted without copying the path and flipping back to bash.

**Comment by luispabon** [comment]
Transparent encrypt/decrypt would be swell :+1:

**Comment by jtognazzi** [comment]
I haven't try, but here is a workaround
https://iuliusathome.blogspot.ch/2016/08/intellij-ansible-vaults-and-windows.html

**Comment by foozmeat** [comment]
I've done something similar to the abov:

Prefs->Tools->External Tools->New

Name: Vault Edit
Group: Ansible
Program: <path to script.sh>
parameters: `edit $FileName$`
working dir: `$FileDir`

script.sh:
```
#!/bin/sh

EDITOR=charm
ansible-vault --vault-password-file=<path to password file> "$@"
```

Assuming you have the pycharm cli tool installed this will open the decrypted vault in a new tab.

**Comment by Ivru** [comment]
+10

**Comment by nitay** [comment]
@vermut any update on this? would love to have it as well

**Comment by lbajsarowicz** [comment]
Hi! To begin with - I'd like to say "Thanks" for the solutions provided before :+1: 

Vault opens successfully and... closes in less than second. When I add sleep do bas script - this works.

How to force ansible-vault to wait for "save" action from PHPStorm?

**Comment by smehan** [comment]
+1 for built in support in the IDE.

**Comment by snezhkov** [comment]
+1

**Comment by advena** [comment]
+1
